Taxation (#270)
MewsSystems/gitbook-connector-api: guidelines/taxation.md

# Taxation

> ## Legal environments and Tax codes

Reviewer: ```suggestion
## Legal environments, tax environments, and tax codes in Mews
```

Reviewer 2: Why T as capital? :)

`## Legal environments and tax codes`

> Each legal environment includes a different set of hard coded tax codes in Mews. Tax codes are used to allow Mews to integrate items’ amounts and tax breakdowns in relation to the received net/gross amounts accurately. If no tax codes are used, then Mews will not calculate the tax breakdown. It expected from integrations to forward the correct tax codes in combination with net or gross amounts.  
> 
> Note that tax environments are subject to change depending on the local government’s ruling. It is expected that integration partners act independently and pull updated information from Mews. In turn, Mews will keep the legal environment and tax codes up to date. 
> 

Reviewer: ```suggestion

Each property (enterprise) in Mews is created with a legal and tax environment corresponding to the region in which it is located. This cannot be changed once the property has been created in Mews' database. 

Each tax environment contains its own set of hard-coded tax codes that are applicable for accounting items generated in the property with that legal environment. The tax codes allow Mews to record and calculate accurately and consistently the tax information associated with each accounting item (if tax should be applied). If no tax codes are used, then Mews will not calculate the tax breakdown. 

It is important to note the following when handling taxable items:

Mews expects its integration partners to send the correct tax codes in combination with net or gross amounts for items that are taxed.

Depending on the local government's ruling, tax environments can be subject to change ((e.g. Covid tax reductions)) from time to time. It is expected that integration partners act independently and pull updated information from Mews. In turn, Mews will keep the legal environment and tax codes up to date. 

```

> ## Implementation of Tax codes 

Reviewer: ```suggestion
## Implementation of tax codes 
```

> The integration should use the following endpoints to retrieve a property’s tax codes.  

Reviewer: ```suggestion
The integration partner should use the following endpoints to correctly retrieve the set of tax codes applicable to a property.  
```

> * [configuration/get](../operations/configuration.md#get-configuration) to retrieve `TaxEnvironmentsCode` that applies to the relevant property.This is needed for [taxEnvironments/getAll](../operations/configuration.md#get-all-tax-environments).
> 
> * [taxEnvironments/getAll](../operations/configuration.md#get-all-tax-environments) to retrieve all `TaxationCodes` within the relevant tax environment of the property. This is needed for [taxations/getAll](../operations/configuration.md#get-all-taxations).
> 
> * [taxations/getAll](../operations/configuration.md#get-all-taxations) to retrieve the individual TaxRatesCodes and their `Value`. 
> 

Reviewer: ```suggestion
1. Call [Get configuration](../operations/configuration.md#get-configuration) to retrieve that `TaxEnvironmentCode` that applies to the relevant property. Note this information for the subsequent API call.

2. Call [Get all tax environments](../operations/configuration.md#get-all-tax-environments) and use the `TaxEnvironmentCode` retrieved earlier to find the corresponding list of `TaxationCodes`. The list of `TaxationCodes` is needed to obtain for the relevant taxation information for each code in the subsequent API call. 

3. Call [Get all taxations](../operations/configuration.md#get-all-taxations) to retrieve the individual tax `Codes`, their names, and values. 

```

## Tax exempt items

> Tax exempt items should include the tax code value `null`. This will lead to Mews not including any tax breakdown. Legal environments in which no actual 0% tax rate exists do not have a 0% tax code in Mews.  

Reviewer: ```suggestion
Tax-exempt items should be sent with `TaxCode` value `null`. Mews in turn not calculate any taxation amount for the item. Legal environments in which no actual 0% tax rate exists do not have a corresponding 0% tax code in Mews.
```

## Multi-tax environment

> Multi-tax environments have instances in which a combination of taxes is applied. In those cases, Mews does not use a single composite tax code but expects that all relevant tax codes are forwarded.  

Reviewer: ```suggestion
Multi-tax environments have instances in which a combination of taxes is applied. In those cases, Mews does not use a single composite tax code but expects all tax codes that apply to the accounting item to be sent to Mews.  
```

> ```javascript
> "TaxCodes": [ 
>     "US-MA-S, 
>     US-MA-EXCISE-0.75%" 
> ]
> ```
> 

Reviewer: ```suggestion
```javascript
"TaxCodes": [ 
    "US-MA-S", 
    "US-MA-EXCISE-0.75%" 
]
```

```

## Validation

> To understand for which time period the specific `TaxRatesCodes` are valid, you would need to review the [taxEnvironments/getAll response](../operations/configuration#response-3). 

Reviewer: ```suggestion
To understand for which time period a set of tax codes are valid in a given tax environment, please review the the validity time stamps for the relevant tax environment in [Get all tax environments](../operations/configuration#response-3). 
```

> # Taxation

Reviewer 3: Not sure this belongs to guidelines. Guidelines are meant as technical guidelines if you check out other content there.

> ## Legal environments, tax environments, and tax codes in Mews

Reviewer 3: Why do we mention legal environments here if the rest of the article doesn't refer to them at all? Or actually, it refers to them but it seems to be mixing legal envs. and tax envs.

> Each property (enterprise) in Mews is created with a legal and tax environment corresponding to the region in which it is located. This cannot be changed once the property has been created in Mews' database. 

Reviewer 3: Shouldn't we stick with a single word - enterprise throughout the whole documentation? Also, let's not mention database, that's a technical detail, I think "This cannot be changed once the enterprise has been created in Mews." works fine.

> Each tax environment contains its own set of hard-coded tax codes that are applicable for accounting items generated in the property with that legal environment. The tax codes allow Mews to record and calculate accurately and consistently the tax information associated with each accounting item (if tax should be applied). If no tax codes are used, then Mews will not calculate the tax breakdown. 

Reviewer 3: I'd omit the "hard-coded" because it has no value for outside readers - they cannot distinguish whether it's hardcoded or not.

> It is important to note the following when handling taxable items:

Reviewer 3: If you lead with `:` then I'd expect a list to follow.

Depending on the local government's ruling, tax environments can be subject to change ((e.g. Covid tax reductions)) from time to time. It is expected that integration partners act independently and pull updated information from Mews. In turn, Mews will keep the legal environment and tax codes up to date. 

## Implementation of tax codes 

The integration partner should use the following endpoints to correctly retrieve the set of tax codes applicable to a property.  

1. Call [Get configuration](../operations/configuration.md#get-configuration) to retrieve that `TaxEnvironmentCode` that applies to the relevant property. Note this information for the subsequent API call.

2. Call [Get all tax environments](../operations/configuration.md#get-all-tax-environments) and use the `TaxEnvironmentCode` retrieved earlier to find the corresponding list of `TaxationCodes`. The list of `TaxationCodes` is needed to obtain for the relevant taxation information for each code in the subsequent API call. 

3. Call [Get all taxations](../operations/configuration.md#get-all-taxations) to retrieve the individual tax `Codes`, their names, and values. 

## Tax exempt items

> Tax-exempt items should be sent with `TaxCode` value `null`. Mews in turn not calculate any taxation amount for the item. Legal environments in which no actual 0% tax rate exists do not have a corresponding 0% tax code in Mews.

Reviewer 3: "Mews in turn not calculate" - that is IMHO grammatically incorrect.

> ## Multi-tax environment

Reviewer 3: For clarity, I think we should call this multi-taxation.

Multi-tax environments have instances in which a combination of taxes is applied. In those cases, Mews does not use a single composite tax code but expects all tax codes that apply to the accounting item to be sent to Mews.  

> ```javascript

Reviewer 3: The code block has no ending.

Reviewer: that was fixed in https://github.com/MewsSystems/gitbook-connector-api/pull/274

# Taxation

## Legal environments, tax environments, and tax codes in Mews

> 

Reviewer 2: ```suggestion
```

Depending on the local government's ruling, tax environments can be subject to change ((e.g. Covid tax reductions)) from time to time. It is expected that integration partners act independently and pull updated information from Mews. In turn, Mews will keep the legal environment and tax codes up to date. 

## Implementation of tax codes 

The integration partner should use the following endpoints to correctly retrieve the set of tax codes applicable to a property.  

1. Call [Get configuration](../operations/configuration.md#get-configuration) to retrieve that `TaxEnvironmentCode` that applies to the relevant property. Note this information for the subsequent API call.

2. Call [Get all tax environments](../operations/configuration.md#get-all-tax-environments) and use the `TaxEnvironmentCode` retrieved earlier to find the corresponding list of `TaxationCodes`. The list of `TaxationCodes` is needed to obtain for the relevant taxation information for each code in the subsequent API call. 

3. Call [Get all taxations](../operations/configuration.md#get-all-taxations) to retrieve the individual tax `Codes`, their names, and values. 

> 

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

## Tax exempt items

Tax-exempt items should be sent with `TaxCode` value `null`. Mews in turn not calculate any taxation amount for the item. Legal environments in which no actual 0% tax rate exists do not have a corresponding 0% tax code in Mews.

## Multi-tax environment

Multi-tax environments have instances in which a combination of taxes is applied. In those cases, Mews does not use a single composite tax code but expects all tax codes that apply to the accounting item to be sent to Mews.  

```javascript
"TaxCodes": [ 
    "US-MA-S", 
    "US-MA-EXCISE-0.75%" 
]

> ## Validation

Reviewer 2: ```suggestion

## Validation
```
